### ISSUE TITLE
'Question: [NativeModules has always been "{}"]'

### ISSUE THREAD

**Issue by KarlLi-dev** [microsoft/react-native-macos#503]
I added a native-module according to:
https://reactnative.dev/docs/native-modules-ios

here is the"CalendarManager" class:
<img width="628" alt="Screen Shot 2020-07-14 at 6 52 17 PM" src="https://user-images.githubusercontent.com/63900685/87417760-3b480a80-c603-11ea-974b-3acc0f4dad14.png">

<img width="439" alt="Screen Shot 2020-07-14 at 6 52 22 PM" src="https://user-images.githubusercontent.com/63900685/87417787-413deb80-c603-11ea-9bba-881c35a6a345.png">

I never get "CalendarManager" in the JS side because "NativeModules" is always "{}".
I printed "NativeModules" in the App.js file.

<img width="580" alt="Screen Shot 2020-07-14 at 6 51 26 PM" src="https://user-images.githubusercontent.com/63900685/87417623-0fc52000-c603-11ea-9ad6-c3559a8c43dd.png">

This is the console result:
<img width="361" alt="Screen Shot 2020-07-14 at 6 54 00 PM" src="https://user-images.githubusercontent.com/63900685/87417942-7c401f00-c603-11ea-9590-031b107cc04e.png">

**Comment by tido64** [comment]
Can you try `RCT_EXPORT_MODULE(CalendarManager)` without the quotes?

**Comment by tom-un** [comment]
@tido64 is correct:
You may specify:
`RCT_EXPORT_MODULE();`
which will use the name of your class or you may specify:
`RCT_EXPORT_MODULE(CalendarManager);`
without quotes to name the module whatever you want.   

But a quoted string literal will not work.   The macro uses the `#` "Stringizing" operator on the argument, so I believe that means if you specify `@"CalendarManagaer"` its expanding to:
```objc
+ (NSString *)moduleName { return @"@\"CalendarManager\""; };
```

**Comment by KarlLi-dev** [comment]
@tom-un  @tido64   Thanks a lot!

I changed "CalendarManager.m" to this 
<img width="436" alt="Screen Shot 2020-07-15 at 9 51 42 AM" src="https://user-images.githubusercontent.com/63900685/87495509-295a7c00-c684-11ea-9e89-68d37874ebf1.png">

and added some Objective-C code in "AppDelegate.m" to check whether "CalendarManager" is initialized on the OC side.
<img width="829" alt="Screen Shot 2020-07-15 at 9 52 13 AM" src="https://user-images.githubusercontent.com/63900685/87495536-34151100-c684-11ea-960f-9dc589811943.png">

But the "NativeModules" of js-side is still "{}".
Here is the result:
<img width="514" alt="Screen Shot 2020-07-15 at 10 05 24 AM" src="https://user-images.githubusercontent.com/63900685/87495678-8f470380-c684-11ea-8747-637998982c67.png">

I initialized the environment by these instructions:

`npx react-native init rnd --version ^0.62.2

cd rnd

npx react-native-macos-init

cd macos && pod install && cd ..

npx react-native run-macos`

Here is my modifications to react-native-macos:
[modifications.zip](https://github.com/microsoft/react-native-macos/files/4922490/modifications.zip)

My Xcode version is 11.5 and MacOS version is 10.15.5.

**Comment by tom-un** [comment]
I'm actually not sure what's going in with the console.log() output.   I get the same thing: NativeModules logs as an empty object in the metro bundler output.    However, if I run in the Chrome debugger and put a breakpoint, the debugger shows all the key/values in NativeModules as expected.  :shrug:

**Comment by KarlLi-dev** [comment]
@tom-un Thanks

To ensure that there's nothing wrong with my development environment, I set up a react-native-ios project and it works perfectly on my machine.

Then I focused on react-native-macos again.
This time I added a test log in this file:
"{project_root}/node_modules/react-native/Libraries/BatchedBridge/NativeModules.js"

<img width="963" alt="Screen Shot 2020-07-17 at 10 51 19 AM" src="https://user-images.githubusercontent.com/63900685/87744629-87bb6200-c81e-11ea-94c1-6ce8f852e3fe.png">

This log does not appear in the console, it seems that 'NativeModules.js' is not even invoked.
I added the same log in the react-native-ios project and it did appear.

This is weird.
